proposals/lip-0055.md

```diff
@@ -66,7 +67,16 @@ As an example, blockchains created with the Lisk SDK that implement the [Random
 
 The schema for the block assets allows each module to include its serialized data individually, which makes the inclusion of module data very flexible. Each module can insert a single entry in the assets. This entry is an object containing a `moduleID` property, indicating the ID of the module handling it, and a generic `data` property that can contain arbitrary serialized data.
 
-Each entry of the block assets is then inserted in a Merkle tree, whose root is included in the block header as the `assetsRoot` property. Inserting the assets root rather than the full assets allows to bound the size of the block header while still authenticating the content of the block assets.
+Each entry of the block assets is then inserted in a Merkle tree, whose root is included in the block header as the `assetRoot` property. Inserting the assets root rather than the full assets allows to bound the size of the block header while still authenticating the content of the block assets.
+
+
+### Failing and Invalid Transactions
+
+In the current protocol, a transaction that is included in a block always has its intended consequences. In particular, if an error occurs during its processing as part of the block generation, the transaction is simply invalid and it is not included in the block payload.
+
+With this LIP, we introduce failing transactions. A failing transaction is one for which an error occurs but which is nevertheless included in a block. 
```

Comment:
I think we should add a more precise definition of valid/invalid and fail here, for instance:

> A transaction is valid, if the hooks `command verification`, `before command execution` and `after command execution` associated to the transaction are executed successfully without error. Otherwise, a transaction is called invalid. Only valid transactions should be added to a block during the block generation, as an invalid transaction makes the block invalid meaning that it would be discarded by any node in the network.

> A valid transaction is executed successfully, if additionally the command execution hook is executed successfully without errors. A valid transaction fails, on the other hand, if an error occurs during the command execution. In this case, all state transitions of the command execution hook are reverted. This means that the transaction has no effect except for those defined in before command execution and after command execution.

Reply:
Note that we also do not say anything about which state transitions are reverted. This should also maybe added to the specifications below.

Reply:
edited in this direction